Source/Core/DiscIO/DirectoryBlob.cpp

```diff
@@ -695,13 +711,12 @@ void DirectoryBlobReader::SetPartitionHeader(DirectoryBlobPartition* partition,
   u64 ticket_size;
   if (wrapped_partition)
   {
-    const auto& ticket = m_wrapped_volume->GetTicket(*wrapped_partition).GetBytes();
-    auto& new_ticket = m_extra_data.emplace_back(ticket);
+    auto& new_ticket =
+        const_cast<std::vector<u8>&>(m_wrapped_volume->GetTicket(*wrapped_partition).GetBytes());
     if (new_ticket.size() > WII_PARTITION_TICKET_SIZE)
       new_ticket.resize(WII_PARTITION_TICKET_SIZE);
     ticket_size = new_ticket.size();
-    m_nonpartition_contents.AddReference(partition_address + WII_PARTITION_TICKET_ADDRESS,
-                                         new_ticket);
+    m_nonpartition_contents.Add(partition_address + WII_PARTITION_TICKET_ADDRESS, new_ticket);
```

Comment:
If you replace the `const_cast` with making a copy, you'll be able to use `std::move` for efficiency when calling Add. (Same for the other two ones below.)